vlasiator.cpp

```diff
@@ -277,6 +309,10 @@ int main(int argn,char* args[]) {
    MPI_Comm comm = MPI_COMM_WORLD;
    MPI_Comm_rank(comm,&myRank);
    
+   if (myRank == MASTER_RANK) {
```

Comment:
I'd put this inside the `if`-clause on line 272, right at the end after the `#endif`

Reply:
Or can I move that before `MPI_Init_thread()`??

Reply:
Ah right, I was focusing just on where the mpiioMessage is generated. I guess it's fine as is then, not worth adding a check for the length of the mpiioMessage or a comment about mpiioMessages as it's easy enough to see what it's associated with.

Reply:
I just saw that we have two calls to MPI_Comm_rank (well the second one is in the unlikely case `if (required > provided){` – can't remember that ever happening), I simplified that in the upcoming commit.

Reply:
Now it's simpler.